Content/ReleaseBriefings/4_6/Section_NewFeatures.tex

```diff
@@ -0,0 +1,156 @@
+%==========================================================================
+
+\begin{frame}[fragile]
+
+  {\Huge Feature highlights}
+
+  \vspace{10pt}
+
+\end{frame}
+
+%==========================================================================
+
+% Examples
+
+% note: always keep the [fragile] for your frames!
+
+%\begin{frame}[fragile]{Example list}
+%  \begin{itemize}
+%      \item Item 1
+%      \item Item 2 with some \texttt{code}
+%      \begin{itemize}
+%        \item Sub-item 2.1
+%        \item Sub-item 2.2
+%      \end{itemize}
+%  \end{itemize}
+%\end{frame}
+
+%\begin{frame}[fragile]{Example code}
+%    \begin{code}[keywords={std}]
+%        #include <iostream>
+%        
+%        int main() {
+%            std::cout << "hello world\n";
+%        }
+%    \end{code}
+%\end{frame}
+
+%\begin{frame}[fragile]{Example table}
+%    \begin{center}
+%        \begin{tabular}{l|l}
+%            a & b \\\hline
+%            c & d
+%        \end{tabular}
+%    \end{center}
+%\end{frame}
+
+%==========================================================================
+
+% \begin{frame}[fragile]\label{sec:new_features}
+
+  % {\Huge Kokkos::Graph features}
+
+  % \vspace{10pt}
+
+% \end{frame}
+
+\begin{frame}[fragile]{Kokkos::Graph recap}
+ \begin{itemize}
+     \item describes asynchronous workloads organised as a direct acyclic graph (DAG)
+     \item executed using \texttt{submit()}, possibly many times, observing dependencies
+      \begin{code}[keywords={auto}]
+auto graph = Kokkos::create_graph([&](auto root) {
+    auto node_A = root.then_parallel_for("A", ...policy..., ...functor...);
+    auto node_B = node_A.then_parallel_for("B", ...policy..., ...functor...);
+    auto node_C = node_A.then_parallel_for("C", ...policy..., ...functor...);
+
+    auto node_D = Kokkos::when_all(node_B, node_C).
+                  then_parallel_for("D", ...policy..., ...functor...);
+});
+
+graph.instantiate();
+
+graph.submit();
+      \end{code}
+ \end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]{Kokkos::Graph new features}
+ \begin{itemize}
+  \item \href{https://github.com/kokkos/kokkos/pull/7629}{\texttt{then} node}: executes a callable on device
+   \item Executed in the \texttt{ExecutionSpace} the graph is submitted to
+     \begin{code}[keywords={auto}]
+auto graph = Kokkos::create_graph([&](auto root) {
+    auto node_A = root.then_parallel_for("A", ...policy..., ...functor...);
+    auto node_B = node_A.then(...functor...);
+});
+
+graph.instantiate();
+
+graph.submit();
+     \end{code}
+ \end{itemize}
+\end{frame}
+
+\begin{frame}[fragile]{Kokkos::Graph new features}
+ \begin{itemize}
+   \item \href{https://github.com/kokkos/kokkos/pull/7664}{interoperability}: create a \texttt{Kokkos::Graph} from a native Cuda/HIP/Sycl graph
+   \item Does not yet allow linking Kokkos nodes to native nodes
```

Comment:
@romintomasetti what does  that mean "linking" nodes

Reply:
It's @JBludau 's wording 😉 But I guess
> It is not yet possible adding nodes created using the native API to a `Kokkos::Graph`

or similar could be an alternative.

Reply:
it is taken from the pr description in https://github.com/kokkos/kokkos/pull/7664

Reply:
Indeed!

Reply:
done in https://github.com/kokkos/kokkos-tutorials/pull/125/commits/1dbcacf143a63ad226a68b91dc3c8ad39c18258a